cli/factory.rs

```diff
@@ -397,14 +397,46 @@ impl CliFactory {
       let workspace_directory = workspace_factory.workspace_directory()?;
       let maybe_external_import_map =
         self.workspace_external_import_map_loader()?.get_or_load()?;
-      CliOptions::from_flags(
+      match CliOptions::from_flags_current_version(
         &self.sys(),
         self.flags.clone(),
         workspace_factory.initial_cwd().clone(),
         maybe_external_import_map,
         workspace_directory.clone(),
-      )
-      .map(Arc::new)
+      ) {
+        Ok(cli_options) => Ok(Arc::new(cli_options)),
+        Err(crate::args::ReadCurrentVersionError::Other(err)) => Err(err),
+        Err(crate::args::ReadCurrentVersionError::NeedsUpgrade) => {
+          let provider = self.npm_registry_info_provider()?.clone();
```

Comment:
if we need to upgrade the lockfile (which requires npm registry info, and therefore is async) spin up a tokio runtime and do the upgrade.

originally i just made this factory function async, but it led to dozens of functions needing async/await